### ISSUE TITLE
Magento 2.3.2 upgrade schema error with Percona XtraDB Cluster

### ISSUE THREAD

**Issue by alchemyconsulting** [magento/magento2#23581]
### Preconditions (*)

1. Magento 2.3.2 (Upgrading from 2.2.9)
2. Centos 7.6.1810, PHP 7.1.30-3, Percona XtraDB Cluster 5.7.26-31

### Steps to reproduce (*)

1. After doing `composer update` to magento 2.3.2, perform `magento setup:upgrade`
2.

### Expected result (*)

1. Expected clean upgrade of modules and database schema
2.

### Actual result (*)

1. 
```
[root@xxx /xxx/html]# bin/magento setup:upgrade
Cache cleared successfully
File system cleanup:
/xxx/html/generated/code/Composer
/xxx/html/generated/code/MageWorx
/xxx/html/generated/code/Magento
/xxx/html/generated/code/Mirasvit
/xxx/html/generated/code/Symfony
The directory '/xxx/html/generated/metadata/' doesn't exist - skipping cleanup
/xxx/html/pub/static/_cache
/xxx/html/pub/static/adminhtml
/xxx/html/pub/static/deployed_version.txt
/xxx/html/pub/static/frontend
/xxx/html/var/view_preprocessed/pub
Updating modules:
Schema creation/updates:
SQLSTATE[HY000]: General error: 1105 Percona-XtraDB-Cluster prohibits changing storage engine of a table (xxx.catalog_product_index_price_cfg_opt_agr_tmp) from transactional to non-transactional with pxc_strict_mode = ENFORCING or MASTER, query was: ALTER TABLE `catalog_product_index_price_cfg_opt_agr_tmp` ENGINE='memory'
```
2.

**Comment by engcom-Charlie** [comment]
Hello @alchemyconsulting  
 We can not reproduce this issue follow steps that you provide.
Please refer to the [Community Forums](https://community.magento.com/) or the [Magento Stack Exchange](https://magento.stackexchange.com/),  for advice or general discussion about this issue.

**Comment by opensourcelib** [comment]
Maybe I'm little late about this issue but just in case someone gets this problem in the future this will be "helpful".
According to Galera Cluster this issued is part of the  Known Limitation to be able to run that query you need to change the value of the variable "pxc_strict_mode" to DISABLED or PERMISSIVE  for more details please read    [Percona website](https://www.percona.com/doc/percona-xtradb-cluster/LATEST/features/pxc-strict-mode.html) 

Also you can read [MariaDB Explanation](https://mariadb.com/kb/en/mariadb-galera-cluster-known-limitations/) I know you are using Percona cluster but MariaDB use the same library for its cluster so that explanation is helpful 

in conclusion you can execute the following query in a running cluster:
`mysql> SET GLOBAL pxc_strict_mode=PERMISSIVE;`

**Comment by alchemyconsulting** [comment]
@opensourcelib - Changing the PXC mode to permissive is generally a Bad Idea (TM) - it turns off the validation and integrity checks that are there for a reason, when you have semi-synchronous replication.
In this particular case I think it means that the relevant table will then not be replicated, so if you then use another node you may have inconsistent/unreliable data.
